kimchi/src/prover.rs

```diff
@@ -108,172 +109,24 @@ where
     runtime_second_col_d8: Option<Evaluations<F, D<F>>>,
 }
 
-impl<G: CommitmentCurve> ProverProof<G>
-where
-    G::BaseField: PrimeField,
-{
-    /// This function constructs prover's zk-proof from the witness & the ProverIndex against SRS instance
-    pub fn create<
-        EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
-        EFrSponge: FrSponge<G::ScalarField>,
-    >(
-        groupmap: &G::Map,
-        witness: [Vec<G::ScalarField>; COLUMNS],
+impl<G: CommitmentCurve + ark_ec::AffineCurve<ScalarField = F>, F: PrimeField> LookupContext<G, F> {
+    pub fn create<EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>>(
```

Comment:
The idea behind this is that we need to be able to build a lookup context to test the foreign field gates fully. This gives us the interface to do that.

Reply:
Yeah-- as @mrmr1993 mentioned above, for now at least, other places in the code need to be able to create lookup contexts. For example, we need to create a lookup context when testing the foreign field gates.

Unfortunately, the code for creating a `LookupContext` is located in the middle of the `Prover`'s `create_recursive` function, which is a single function that is over 1000 lines of code long.

Since the `LookupContext` is a self-contained thing that is created in a regular way, @mrmr1993 and I wanted to encapsulate this logic into a `LookupContext::create()` function.  This will allow other parts of the code that need a lookup context to get one, with the added bonus of `create_recursive` being a bit smaller and a bit more modular.

It's not perfect and there is surely more (and better) refactoring that can be done.  I think your idea to extract the different parts of protocol login into their own individual functions is a good one, but this change is not about doing that and it doesn't preclude those future improvements.  This is just an intermediate change, so that we are able create a `LookupContext` for our present needs.